plugins/modules/aci_qos_dot1p_class.py

```diff
@@ -0,0 +1,347 @@
+    ACIModule,
+    aci_argument_spec,
+    aci_annotation_spec,
+    aci_owner_spec,
+    aci_contract_dscp_spec,
+)
+from ansible_collections.cisco.aci.plugins.module_utils.constants import MATCH_TARGET_COS_MAPPING
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        qos_custom_policy=dict(type="str", aliases=["qos_custom_policy_name"]),
+        priority=dict(
+            type="str",
+            choices=[
+                "level1",
+                "level2",
+                "level3",
+                "level4",
+                "level5",
+                "level6",
+                "unspecified",
+            ],
+            aliases=["prio"],
+        ),
+        dot1p_from=dict(type="str", choices=list(MATCH_TARGET_COS_MAPPING.keys())),
+        dot1p_to=dict(type="str", choices=list(MATCH_TARGET_COS_MAPPING.keys())),
+        dot1p_target=aci_contract_dscp_spec(),
+        target_cos=dict(type="str", choices=list(MATCH_TARGET_COS_MAPPING.keys())),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "qos_custom_policy", "dot1p_from", "dot1p_to"]],
+            ["state", "present", ["tenant", "qos_custom_policy", "dot1p_from", "dot1p_to"]],
+        ],
+    )
+
+    tenant = module.params.get("tenant")
+    qos_custom_policy = module.params.get("qos_custom_policy")
+    priority = module.params.get("priority")
+    dot1p_from = MATCH_TARGET_COS_MAPPING.get(module.params.get("dot1p_from"))
+    dot1p_to = MATCH_TARGET_COS_MAPPING.get(module.params.get("dot1p_to"))
+    dot1p_target = module.params.get("dot1p_target")
+    target_cos = MATCH_TARGET_COS_MAPPING.get(module.params.get("target_cos"))
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="qosCustomPol",
+            aci_rn="qoscustom-{0}".format(qos_custom_policy),
+            module_object=qos_custom_policy,
+            target_filter={"name": qos_custom_policy},
+        ),
+        subclass_2=dict(
+            aci_class="qosDot1PClass",
+            aci_rn="dot1P-{0}-{1}".format(dot1p_from, dot1p_to),
+            module_object=qos_custom_policy,
+            target_filter={"from": dot1p_from, "to": dot1p_to},
+        ),
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        aci.payload(
+            aci_class="qosDot1PClass",
+            class_config={
```

Comment:
Shall we use the dict() keyword here instead for consistency?

Reply:
class_config = dict(
    prio=priority,
    from_=dot1p_from,
    to=dot1p_to,
    target=dot1p_target,
    targetCos=target_cos,
)